apps/backend/src/modules/chats/chat.repository.ts

```diff
@@ -126,9 +132,41 @@ class ChatRepository implements Repository<ChatEntity> {
 			)
 			.whereExists(ChatModel.relatedQuery(RelationName.LAST_MESSAGE))
 			.andWhere((builder) => {
-				void builder
-					.where({ firstUserId: userId })
-					.orWhere({ secondUserId: userId });
+				if (search) {
+					void builder
+						.where((builder) => {
+							void builder
+								.where(
+									"firstUser:userDetails.firstName",
+									"ilike",
+									`%${search}%`,
+								)
+								.orWhere(
+									"firstUser:userDetails.lastName",
+									"ilike",
+									`%${search}%`,
+								);
+						})
+						.andWhere({ secondUserId: userId })
+						.orWhere((builder) => {
+							void builder
+								.where(
```

Comment:
```suggestion
								.whereILike(
```
And I think we can do this without `if ... else` statement, because if `search` is empty, then `.whereILike("column", "%%")` will return everything

Reply:
Yes, it works without ``if ... else``. Thanks

Reply:
@fshabanov I agree that query `where field_name ilike '%%'` (with empty search template) will give the same result as query without that `where` statement. But I checked it in the execution plan and it is different from query without `where` statement. It is really filtering rows by excluding null values (`where field_name ilike '%%'` is equivalent to `where field_name is not null`. So I think query with empty `'%%'` is less efficient than query without ilike at all. In time it is 1.5 - 2 times slower than query without filtering. I checked it on a simple query with small row count, but I think that on chat querying (as it is on our situation) it will be more notable. Is it worth it?